spec/requests/tasks_spec.rb

```diff
@@ -0,0 +1,51 @@
+describe 'TasksController' do
+  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+
+  def expect_deleted(*args)
+    args.each do |arg|
+      expect(MiqTask.find_by(:id => arg.id)).to be_nil
+    end
+  end
+
+  it 'deletes on DELETE' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete, :delete)
+
+    url = api_task_url(nil, task.id)
+
+    delete(url)
+
+    expect(response).to have_http_status(204)
+    expect_deleted(task)
+  end
+
+  it 'deletes on POST' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete)
+
+    url = api_task_url(nil, task.id)
+    data = {:action => 'delete'}
+
+    post(url, :params => data)
+
+    expect(response).to have_http_status(200)
+    expect_deleted(task)
+  end
+
+  it 'bulk deletes' do
+    api_basic_authorize collection_action_identifier(:tasks, :delete)
+
+    url = api_tasks_url
+    data = {
+      :action    => 'delete',
+      :resources => [
+        {:href => api_task_url(nil, task.id)},
+        {:href => api_task_url(nil, task2.id)},
+      ],
+    }
+
+    post(url, :params => data)
+
+    expect(response).to have_http_status(200)
+    expect_deleted(task, task2)
```

Comment:
Generally, the API is concerned with the responses rather than validating the objects, something like [this](https://github.com/ManageIQ/manageiq-api/blob/1e91ed2b7ba2d20726ac8d76cdeb69e7bc8f7f3d/spec/requests/custom_button_sets_spec.rb#L136-L141)

Reply:
Well, I don't care about the response at all, only that it actually deleted the thing :).

Should I add something like that then?

But.. more and more this looks like I'm testing your generic functions for one specific case, which is identical to all the other specific cases.

Would you be interested in specs for the generic `resource_delete` instead?

Reply:
Or at least a set of helpers so that we can write the tests once, and then only provide that things that should be different?

That task_spec should not really have to be more code than something like..

```
describe 'TasksController' do
  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }

  test_deletion(:methods => [:delete, :post, :bulk], :resources => [:task, :task2])
end
```